doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+# Operation\<T>
+
+The Azure SDK for C++ defines an Operation as an abstract class to support a long running operation ([LRO](http://restalk-patterns.org/long-running-operation-polling.html)).
```

Comment:
Rather than external (http) link, maybe we link to something like this instead:
https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#13-long-running-operations